catalog/catalog.go

```diff
@@ -0,0 +1,64 @@
+package catalog
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ghodss/yaml"
+	"github.com/giantswarm/microerror"
+	"github.com/levigross/grequests"
+)
+
+// GetLatestChart returns the latest chart tarball file in the specified catalog.
+func GetLatestChart(ctx context.Context, catalog, app string) (string, error) {
+	index, err := getIndex(catalog)
```

Comment:
Yes that makes sense to me. We could have it in https://github.com/giantswarm/appcatalog which currently just has the helm chart used for `opsctl ensure appcatalogs`.

WDYT?

Reply:
👍

Reply:
I will close this PR after all suggestion applied, then move it to `appcatalog` repo.

Reply:
Cool thanks and it will be good to have logic like generating the tarball URLs there instead of app-operator. As then we can reuse that code in e2e and other operators.